components/autocomplete/templates.md

```diff
@@ -12,99 +12,88 @@ position: 20
 
 The AutoComplete component allows you to change what is rendered in its items, header and footer through templates.
 
-List of the available templates:
+>caption In this article:
 
 * [Item Template](#item-template)
-* [Header](#header)
-* [Footer](#footer)
-
+* [Header Template](#header-template)
+* [Footer Template](#footer-template)
+* [No Data Template](#no-data-template)
+* [Example](#example)
 
 ## Item Template
 
-The Item template determines how the individual items are rendered in the dropdown element of the component. By default, the text from the particular suggestions is rendered.
+@[template](/_contentTemplates/dropdowns/templates.md#item-template)
 
->caption Item Template Example
+## Header Template
 
-````CSHTML
-@* Define what renders for the items in the dropdown *@
+@[template](/_contentTemplates/dropdowns/templates.md#header-template)
 
-<TelerikAutoComplete Data="@Suggestions" @bind-Value="@Role" Placeholder="Write your position">
-    <ItemTemplate>
-        Are you a&nbsp;<strong>@context</strong>
-    </ItemTemplate>
-</TelerikAutoComplete>
+## Footer Template
 
-@code{
-    string Role { get; set; }
+@[template](/_contentTemplates/dropdowns/templates.md#footer-template)
 
-    List<string> Suggestions { get; set; } = new List<string> {
-        "Manager", "Developer", "QA", "Technical Writer", "Support Engineer", "Sales Agent", "Architect", "Designer"
-    };
-}
-````
+## No Data Template
 
->caption The result from the code snippet above
+@[template](/_contentTemplates/dropdowns/templates.md#no-data-template)
 
-![Blazor Autocomplete Item Template](images/autocomplete-item-template.png)
+## Example
 
-## Header
-
-The header is content that you can place above the list of items inside the dropdown element. It is always visible when the combobox is expanded. By default it is empty.
-
->caption Header Example
+>caption Using AutoComplete Templates
 
 ````CSHTML
-@* Define a header in the dropdown *@
+@* AutoComplete component with HeaderTemplate, ItemTemplate, FooterTemplate and NoDataTemplate *@
 
-<TelerikAutoComplete Data="@Suggestions" @bind-Value="@Role" Placeholder="Write your position">
+<p>
+    <TelerikCheckBox @bind-Value="@IsDataAvailable" OnChange="@OnCheckBoxChangeHandler" />
+    AutoComplete has data
+</p>
+
+<TelerikAutoComplete Data="@AutoCompleteData" @bind-Value="@Role" Placeholder="Write your position">
     <HeaderTemplate>
         <strong>Write your own if you don't see it in the list</strong>
     </HeaderTemplate>
+    <ItemTemplate>
+        Are you a&nbsp;<strong>@context</strong>
```

Comment:
Does the `&nbsp;` serve a special purpose here?

Reply:
It is not needed, I removed it from teh examples.